commit b670813cf0757f01ce0d863aad429b7342f29a3e
Author: Daniel Martinez <danielmartinez@cock.li>
Date:   Mon Oct 14 15:25:09 2024 -0400

    Add checks around locales
    
    With musl libc, many of the locale macros are not defined, causing build
    errors
    
    Check that the macros are defined before using them

diff --git a/src/main/locale.cpp b/src/main/locale.cpp
index 69ea619..ed84f5d 100644
--- a/modules/lsp-common-lib/src/main/locale.cpp
+++ b/modules/lsp-common-lib/src/main/locale.cpp
@@ -40,14 +40,24 @@ namespace lsp
                 case LC_MONETARY:       flags   = LC_MONETARY_MASK; break;
                 case LC_MESSAGES:       flags   = LC_MESSAGES_MASK; break;
                 case LC_ALL:            flags   = LC_ALL_MASK; break;
-        #if defined(PLATFORM_LINUX)
+        #if defined(LC_PAPER)
                 case LC_PAPER:          flags   = LC_PAPER_MASK; break;
+        #endif
+        #if defined(LC_NAME)
                 case LC_NAME:           flags   = LC_NAME_MASK; break;
+        #endif
+        #if defined(LC_ADDRESS)
                 case LC_ADDRESS:        flags   = LC_ADDRESS_MASK; break;
+        #endif
+        #if defined(LC_TELEPHONE)
                 case LC_TELEPHONE:      flags   = LC_TELEPHONE_MASK; break;
+        #endif
+        #if defined(LC_MEASUREMENT)
                 case LC_MEASUREMENT:    flags   = LC_MEASUREMENT_MASK; break;
+        #endif
+        #if defined(LC_IDENTIFICATION)
                 case LC_IDENTIFICATION: flags   = LC_IDENTIFICATION_MASK; break;
-        #endif /* PLATFORM_LINUX */
+        #endif
 
                 default:
                     return INVALID_LOCALE;
